indexdata.go

```diff
@@ -376,6 +382,28 @@ func (r *ngramIterationResults) candidates() []*candidateMatch {
 func (d *indexData) iterateNgrams(query *query.Substring) (*ngramIterationResults, error) {
 	str := query.Pattern
 
+	if d.bloomNames.Len() > 0 && len(query.Pattern) >= bloomHashMinWordLength {
+		pat := []byte(query.Pattern)
+		fnmatch := d.bloomNames.maybeHasBytes(pat)
```

Comment:
from reading this I still think `maybeHasBytes` returning true if `Len() == 0 || len(pat) < bloomHashMinWordLength` will make this call site much more readable at minimal cost given we normally call maybeHasBytes. If feels like it will reduce potential errors, like below where you don't check if bloomContents is non-empty (even though if one is set, the other is always set as the PR currently stands)

Reply:
I rewrote this for clarity, please have a look.